source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -47,6 +47,9 @@ namespace TcpProxy {
   COUNTER(downstream_flow_control_resumed_reading_total)                                           \
   COUNTER(idle_timeout)                                                                            \
   COUNTER(max_downstream_connection_duration)                                                      \
+  COUNTER(on_demand_cluster_attempt)                                                               \
```

Comment:
Thinking about what @jmarantz might say, should these stats only be created if `OnDemand` is configured, to reduce the amount of stats memory used when the feature isn't being used?

Reply:
I don't have enough context to be sure, but if it appears this would only be adding a stat to TCP Proxy instances. That seems OK to me, as long as it isn't a new stat which we get per-cluster or per-host for all envoy users.

Reply:
That's a per tcp proxy instance. Is the conclusion is that we can leave it as always allocate this counter?

We can always move it optional any time in the future, right?

Reply:
On second thought, looking at the data structure below, could you fold the 3 new stats, plus `odscs_` and `odscs_timeout_` all into one optional structure?

Reply:
Good point! Putting the metric together improves memory footprint and coalition.

Reply:
Now the counters are generated if on demand is enabled